xds/src/main/java/io/grpc/xds/EdsLoadBalancer.java

```diff
@@ -0,0 +1,737 @@
+      endpointWatcher = new EndpointWatcherImpl(
+          helper, pickerFactory, lbRegistry, random, loadStatsStore, orcaPerRequestUtil,
+          orcaOobUtil);
+      xdsClient.watchEndpointData(edsServiceName, endpointWatcher);;
+      this.edsServiceName = edsServiceName;
+    }
+  }
+
+  @Override
+  public void handleNameResolutionError(Status error) {
+    channelLogger.log(ChannelLogLevel.DEBUG, "Name resolution error: '%s'", error);
+    // NO-OP?
+  }
+
+  @Override
+  public boolean canHandleEmptyAddressListFromNameResolution() {
+    return true;
+  }
+
+  @Override
+  public void shutdown() {
+    channelLogger.log(ChannelLogLevel.DEBUG, "EDS load balancer is shutting down");
+
+    if (endpointWatcher != null) {
+      xdsClient.cancelEndpointDataWatch(edsServiceName, endpointWatcher);
+      endpointWatcher.reset();
+    }
+  }
+
+  static final class EdsConfig {
+
+    /**
+     * Name of cluster to query EDS for.
+     */
+    final String name;
+
+    final Object localityPickingPolicy;
+
+    EdsConfig(String name, Object localityPickingPolicy) {
+      this.name = checkNotNull(name, "name");
+      this.localityPickingPolicy = checkNotNull(localityPickingPolicy);
+    }
+  }
+
+  static final class EndpointWatcherImpl implements EndpointWatcher {
+
+    private static final String ROUND_ROBIN = "round_robin";
+    private static final long DELAYED_DELETION_TIMEOUT_MINUTES = 15L;
+    private static final EndpointWatcherImpl.PickerFactory pickerFactoryImpl =
+        new EndpointWatcherImpl.PickerFactory() {
+          @Override
+          public SubchannelPicker picker(List<WeightedChildPicker> childPickers) {
+            return new InterLocalityPicker(childPickers);
+          }
+        };
+
+    private final Helper helper;
+    private final PickerFactory pickerFactory;
+    private final LoadBalancerProvider loadBalancerProvider;
+    private final ThreadSafeRandom random;
+    private final LoadStatsStore loadStatsStore;
+    private final OrcaPerRequestUtil orcaPerRequestUtil;
+    private final OrcaOobUtil orcaOobUtil;
+    private final PriorityManager priorityManager = new PriorityManager();
+    private final Map<Locality, LocalityLbInfo> localityMap = new HashMap<>();
+    // Most current set of localities instructed by traffic director
+    private Set<Locality> localities = ImmutableSet.of();
+    private List<DropOverload> dropOverloads = ImmutableList.of();
+    private long metricsReportIntervalNano = -1;
+
+    EndpointWatcherImpl(
+        Helper helper,
+        PickerFactory pickerFactory,
+        LoadBalancerRegistry lbRegistry,
+        ThreadSafeRandom random,
+        LoadStatsStore loadStatsStore,
+        OrcaPerRequestUtil orcaPerRequestUtil,
+        OrcaOobUtil orcaOobUtil) {
+      this.helper = checkNotNull(helper, "helper");
+      this.pickerFactory = checkNotNull(pickerFactory, "pickerFactory");
+      loadBalancerProvider = checkNotNull(
+          lbRegistry.getProvider(ROUND_ROBIN),
```

Comment:
Why this is getting a hard-coded round-robin child load balancer here? This should come from `EdsConfig`